lib/corebulk.go

```diff
@@ -113,12 +113,16 @@ func (c *Conn) NewBulkIndexer(maxConns int) *BulkIndexer {
 }
 
 func (c *Conn) NewBulkIndexerWithChannelSize(maxConns, bulkChannelSize int) *BulkIndexer {
+	return c.NewBulkIndexerWithParams(maxConns, bulkChannelSize, func() int { return BulkMaxBuffer }, func() int { return BulkMaxDocs })
+}
+
+func (c *Conn) NewBulkIndexerWithParams(maxConns, bulkChannelSize int, bulkMaxBufferCb, bulkMaxDocsCb func() int) *BulkIndexer {
 	b := BulkIndexer{conn: c, sendBuf: make(chan *bytes.Buffer, maxConns)}
 	b.needsTimeBasedFlush = true
 	b.buf = new(bytes.Buffer)
 	b.maxConns = maxConns
-	b.BulkMaxBuffer = BulkMaxBuffer
-	b.BulkMaxDocs = BulkMaxDocs
+	b.BulkMaxBufferCB = bulkMaxBufferCb
```

Comment:
Should we set a default value?  Just to be safe if say 0 is passed in

Reply:
You mean if someone passes `func() int { return 0 }`? The default if you don't specify a callback at all (i.e. if you keep using the `NewBulkIndexerWithChannelSize` constructor that doesn't take a callback) is to [use the existing default](https://github.com/launchdarkly/elastigo/pull/20/files#diff-aa16db2eed67a9c127da15da91d60aadR116) (of 16384 bytes).

Reply:
Oh derp I did not see you had a new function.  M